src/services/sanity.js

```diff
@@ -1301,6 +1310,42 @@ export async function fetchSanity(query, isList) {
   }
 }
 
+async function needsAccessDecorator(results)
+{
+    if (globalConfig.sanityConfig.useDummyRailContentMethods) return results;
+    let userPermissions = await getUserPermissions();
+    userPermissions = new Set(userPermissions);
+    if (Array.isArray(results)) {
+        results.forEach((result) => {
+            result['need_access'] = doesUserNeedAccessToContent(result, userPermissions);
+        });
+    } else {
+        results['need_access'] = doesUserNeedAccessToContent(results, userPermissions);
+    }
+    return results;
+}
+
+function doesUserNeedAccessToContent(result, userPermissions)
```

Comment:
Nitpicking, but is there any sort of structure we could use the the parameters, so that we know `result.permission_id` is valid, etc.?

Reply:
Tested for custom field or if premission_id is empty/null requests `needs_access` will always be true